### ISSUE TITLE
Extra containers are launched for a browser config

### ISSUE THREAD

**Issue by animulli** [zalando-incubator/zalenium#30]
Running this command
`  docker run --rm -ti --name zalenium -p 4444:4444 -p 5555:5555 \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -v /tmp/videos:/home/seluser/videos \
    dosel/zalenium start --sauceLabsEnabled false`

Against a Protractor test with the following config
`exports.config = {
  seleniumAddress: 'http://localhost:4444/wd/hub',
  specs: ['todo1-spec.js', 'todo2-spec.js'],
  multiCapabilities: [{
    'browserName': 'firefox'
  }
  ,{
    'browserName': 'chrome',
    shardTestFiles: true,
    maxInstances: 2
  }
  ],`

Launches 4 additional Firefox containers. Expected only additional Chrome containers. Also, it seems that once the tests are completed, Zalenium doesn't remove any of the stopped containers that were spawned, as well as stopping a couple of the extra Firefox containers.
![image](https://cloud.githubusercontent.com/assets/7917063/20699815/30ea740a-b5be-11e6-83c2-7d1d7c7e8249.png)

**Comment by animulli** [comment]
There are removed once you stop the Zalenium hub though

**Comment by animulli** [comment]
nvm read the MAC notes

**Comment by animulli** [comment]
but i am still noticing that stopped containers aren't removed until the Zalenium hub node is stopped

**Comment by diemol** [comment]
HI @animulli, thanks for trying Zalenium! 

Let me answer you part by part:

> Launches 4 additional Firefox containers. Expected only additional Chrome containers. Also, it seems that once the tests are completed, Zalenium doesn't remove any of the stopped containers that were spawned, as well as stopping a couple of the extra Firefox containers.

When the `--chromeContainers` and the `--firefoxContainers` parameters are not overwritten, Zalenium will start with one Chrome node and one Firefox node. 

The additional Firefox nodes/containers are launched because the logic we have right now is that if tests requests using Firefox are submitted to the hub, probably more will come, so we create some nodes in advance. Same applies for Chrome.

We are not sure if this is the right behaviour, maybe over time we find a different behaviour that makes more sense, or maybe we stay with this one with a few improvements. 

> but i am still noticing that stopped containers aren't removed until the Zalenium hub node is stopped

Yes, this is normal and expected. When a new node/container is created, and it has not been used, we leave it there since we don't know if more test requests will come. The nodes/containers are removed only after one test has been executed.

If you have any more doubts or questions, don't hesitate to create more issues.

**Comment by animulli** [comment]
@diemol ah makes sense now, sounds like a good idea, I did run into timeout issues on some of my protractor tests waiting for a node to start up. I'll try to repro again and create a new issue if i see it again. thanks.
